search/search_quality_tests/search_quality_tests.cpp

```diff
@@ -160,8 +176,163 @@ void CalcStatistics(vector<double> const & a, double & avg, double & maximum, do
   stdDev = sqrt(var);
 }
 
+// Unlike strings::Tokenize, this function allows for empty tokens.
+void Split(string const & s, char delim, vector<string> & parts)
+{
+  istringstream iss(s);
+  string part;
+  while (getline(iss, part, delim))
+    parts.push_back(part);
+}
+
+// Returns the position of the result that is expected to be found by geocoder completeness
+// tests in the |result| vector or -1 if it does not occur there.
+int FindResult(TestSearchEngine & engine, string const & country, uint64_t const featureId,
+               double const lat, double const lon, vector<search::Result> const & results)
+{
+  auto const mwmId = engine.GetMwmIdByCountryFile(platform::CountryFile(country));
+  FeatureID const expectedFeatureId(mwmId, featureId);
+  for (size_t i = 0; i < results.size(); ++i)
+  {
+    auto const & r = results[i];
+    auto const featureId = r.GetFeatureID();
+    if (featureId == expectedFeatureId)
+      return i;
+  }
+
+  // Another attempt. If the queries are stale, feature id is useless.
+  // However, some information may be recovered from (lat, lon).
+  double const kEps = 1e-6;
+  for (size_t i = 0; i < results.size(); ++i)
+  {
+    auto const & r = results[i];
+    if (r.HasPoint() &&
+        my::AlmostEqualAbs(r.GetFeatureCenter(), MercatorBounds::FromLatLon(lat, lon), kEps))
+      return i;
+  }
+  return -1;
+}
+
+// Reads queries in the format
+//   CountryName.mwm:featureId;type;lon;lat;city;street;<housenumber or housename>
+// from |path|, executes them against the |engine| with viewport set to |viewport|
+// and reports the number of queries whose expected result is among the returned results.
+// Exact feature id is expected, but a close enough (lat, lon) is good too.
+void CheckCompleteness(string const & path, m2::RectD const & viewport, TestSearchEngine & engine)
+{
+  ifstream stream(path.c_str());
+  CHECK(stream.is_open(), ("Can't open", path));
+
+  my::Timer timer;
+
+  uint32_t totalQueries = 0;
+  uint32_t malformedQueries = 0;
+  uint32_t expectedResultsFound = 0;
+  uint32_t expectedResultsTop1 = 0;
+
+  // todo(@m) Process the queries on the fly and do not keep them.
+  vector<string> queries;
```

Comment:
It seems that you need to create a structure containing country, id, lat, lon and create a single vector of these structures.